File: processor/src/test/resources/approvals/jpa/IdsProductIdAttributeConverter.java
package test.boundary.persistence.jpa;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import scenarios.java.Ids;

import javax.annotation.processing.Generated;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.JpaGenerator")
@Converter(
    autoApply = true
)
public class IdsProductIdAttributeConverter implements AttributeConverter<Ids.ProductId, String> {
  public String convertToDatabaseColumn(Ids.ProductId type) {
    if(type == null) {
      return null;
    }
    return type.value();
  }

  public Ids.ProductId convertToEntityAttribute(String dbValue) {
    if(dbValue == null) {
      return null;
    }
    return Ids.ProductId.of(dbValue);
  }
}

File: ksp/src/test/resources/approvals/jpa/IdsProductIdAttributeConverter.kt
package test.boundary.persistence.jpa

import jakarta.`annotation`.Generated
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import kotlin.String
import scenarios.kotlin.Ids

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.JpaGenerator")
@Converter(autoApply = true)
public class IdsProductIdAttributeConverter : AttributeConverter<Ids.ProductId?, String?> {
  override fun convertToDatabaseColumn(type: Ids.ProductId?): String? = type?.value

  override fun convertToEntityAttribute(dbValue: String?): Ids.ProductId? = dbValue?.let { Ids.ProductId.of(dbValue) }
}
